packages/api-client/src/views/Request/libs/oauth2.ts

```diff
@@ -16,45 +17,108 @@ export const isOauth2Example = (
 ): example is SecuritySchemeOauth2ExampleValue =>
   example.type.startsWith('oauth')
 
+type PKCEState = {
+  codeVerifier: string
+  codeChallenge: string
+  codeChallengeMethod: string
+}
+
+/**
+ * Generates a random string of specified length using crypto API
+ * Length must be between 43 and 128 characters as per RFC 7636
+ */
+const generateCodeVerifier = (length = 64): string => {
+  const array = new Uint8Array(length)
+  crypto.getRandomValues(array)
+
+  return Array.from(array)
+    .map(
+      (x) =>
+        'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~'[
+          x % 66
+        ],
+    )
+    .join('')
+}
+
+/**
+ * Creates a code challenge from the code verifier using SHA-256
+ */
+const generateCodeChallenge = async (verifier: string): Promise<string> => {
+  const encoder = new TextEncoder()
+  const data = encoder.encode(verifier)
+  const digest = await crypto.subtle.digest('SHA-256', data)
+
+  return btoa(String.fromCharCode(...new Uint8Array(digest)))
+    .replace(/\+/g, '-')
+    .replace(/\//g, '_')
+    .replace(/=+$/, '')
+}
+
 /**
  * Authorize oauth2 flow
  *
  * @returns the accessToken
  */
-export const authorizeOauth2 = (
+export const authorizeOauth2 = async (
   scheme: SecuritySchemeOauth2,
   example: SecuritySchemeOauth2ExampleValue,
   /** We use the active server to set a base for relative redirect uris */
   activeServer: Server,
-) =>
-  new Promise<string>((resolve, reject) => {
+): Promise<ErrorResponse<string>> => {
```

Comment:
I had to change this around anyway to handle some await crypto action so I ended up switching it to the ErrorResponse style for some better error handling